### PR TITLE
New Shuttle UI + Fixes #1327

### LINKED ISSUE #1327: TODO: Shuttle Control UI new elements B: 45
# Criteria: 

### Two new GUI elements are required for the Shuttle UI:
 - Shuttle Co-ord read outs. A Vector2Int of the current world position of the shuttle needs to be displayed on the shuttle UI
 - Fuel meter. Atm we just need a GUI element of a fuel meter. When Plasma fuel feature is added to the shuttles then we will work on its functionality.

A bounty of 45usd will be awarded to the contributor on the successful merge of these features.

------------------
#### Suggested approach:

 1) Find the TabShuttleControl prefab and drag it underneath the UIManager so that you can work on it in the scene
 2) Rearrange the UI so that you have some room to add a co-ordinate read out and a fuel meter
 3) Create the new graphics to support the additional elements and store them in Assets/UI/Shuttle_Controls
 5) All data for the new UI elements should be present on the client (fuel is not ready yet). Modify the GUI_ShuttleControl script to update the co-ordinates based off the WorldPosition of the associated MatrixMove component
 6) Remove the TabShuttleControl prefab from the Managers tree and apply the changes to the prefab (Applying them while it is in the tree will merge the TabShuttleControl with the Managers prefab and it will break everything)
 7) remove the prefab from the hierarchy, jump on a shuttle and test

### PR DESCRIPTION
**Pull request by kaffo**
* Added Fuel Bar (Credit to Foma for the image)
  * Exposed API to set the % fuel remaining for when fuel gets
  implemented
  * Should be scalable/moveable and not break
* Added Coordinates
  * Uses parent MatrixMove to update position from server
* Reworked shuttle UI
  * Used fewer fonts/tried to use similar sizes
  * Moved the shuttle movement controls to the right
  * Much smaller buttons
  * Fixed background on radar
* Bonus UI swap script for Nanotransen vs other agencies
  * Set the UI type on the MatrixMove script and see the UI change

### Purpose
This PR is to fix #1327 
Credit to @fomalsd for his fuel gauge
Let me know if there's anything you want to change, it's pretty simple stuff

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer (with 2 clients, if applicable)

### Notes:
Picture of the UI:
image

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Shuttles/GUI_FuelGauge.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	public RectTransform colourBarTransform;
+	public RectTransform pointerTransform;
+
+	private float fuelGaugeMaxWidth = 0f;
+	
+	void Start()
+    {
+		if (fuelGaugeTransform == null || colourBarTransform == null || pointerTransform == null)
+		{
+			Logger.LogError("No Fuel Gauge Set on Shuttle!");
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Shuttles/GUI_FuelGauge.cs]:
```suggestion
			Logger.LogError("No Fuel Gauge Set on Shuttle!", Category.UI);
```

File: UnityProject/Assets/Scripts/UI/Shuttles/GUI_CoordReadout.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	public GUI_ShuttleControl shuttleControlScript;
+
+	private int valueX = 0;
+	private int valueY = 0;
+
+	void Start()
+    {
+        if (xText == null || yText == null || shuttleControlScript == null)
+		{
+			Logger.LogError("Coord Readout not setup correctly!");
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/UI/Shuttles/GUI_CoordReadout.cs]:
```suggestion
			Logger.LogError("Coord Readout not setup correctly!", Category.UI);
```

**Review by fomalsd — CHANGES_REQUESTED** [review]
Scr 2019-04-06 в 15 43 31

 - don't see the fuel gauge, sprite is white
 - coordinates don't work, at least on syndie shuttle. always shows 100,100
 - disclaimer still says "property of nanotraNsen" on syndie ship
 - you didn't commit meta files for your added scripts and textures. do it!

**Comment by kaffo** [comment]
I resetted my branch and tried it clean and it seems to work with what I've got commited now. So hopefully it should be good now

**Review by fomalsd — APPROVED** [review]

**Comment by fomalsd** [comment]
all good now, minor thing is that elements look too small now (text is kinda hard to read)
Scr 2019-04-07 в 1 19 52

**Comment by fomalsd** [comment]
yep, sizes are good now!

**Pull request merged**